index.js

```diff
@@ -45,7 +45,7 @@ export function validate(origin, args) {
 		].forEach(([n, v, t, o = {}]) => {
 			const r = check(v, t, o);
 
-			if (!condition(r, o)) {
+			if (!(condition(r, o) && o.optional)) {
```

Comment:
Did a goof, primary condition check failed due the property not existing to begin with.

Next intended change is to modify this for use on the check described on cb7874c7078f9dc71aa5889a6f73f4aa8c9bb437